ospfd/ospf_vty.c

```diff
@@ -8558,6 +8558,10 @@ DEFUN (ospf_distance_ospf,
 	VTY_DECLVAR_INSTANCE_CONTEXT(ospf, ospf);
 	int idx = 0;
 
+	ospf->distance_intra = 0;
+	ospf->distance_inter = 0;
+	ospf->distance_external = 0;
+
```

Comment:
This resets all values, which is not clear from the CLI. Here is the allowed forms of the command: 
/permutations "distance ospf {intra-area (1-255)|inter-area (1-255)|external (1-255)}"
  distance ospf intra-area (1-255) inter-area (1-255) external (1-255)
  distance ospf intra-area (1-255) inter-area (1-255)
  distance ospf intra-area (1-255) external (1-255) inter-area (1-255)
  distance ospf intra-area (1-255) external (1-255)
  distance ospf intra-area (1-255)
  distance ospf inter-area (1-255) intra-area (1-255) external (1-255)
  distance ospf inter-area (1-255) intra-area (1-255)
  distance ospf inter-area (1-255) external (1-255) intra-area (1-255)
  distance ospf inter-area (1-255) external (1-255)
  distance ospf inter-area (1-255)
  distance ospf external (1-255) intra-area (1-255) inter-area (1-255)
  distance ospf external (1-255) intra-area (1-255)
  distance ospf external (1-255) inter-area (1-255) intra-area (1-255)
  distance ospf external (1-255) inter-area (1-255)
  distance ospf external (1-255)

If the user configures one value at a time, (intra-area, inter-area, external), it is not going to be clear that this is resetting the other two values. The command structure at least suggest that the user should be able to do so.

Reply:
Ok maybe I'm reading this incorrectly. Every time you set the distance, it is going to be the actual configuration,  you should not be able to "accumulate"  the configuration over multiple runs, right?

Reply:
If the user configures `keyword 10 20 30` and then later does `keyword 10 20` then that has to delete the 30.  Basically what they configure is what needs to show up in the configuration.

Reply:
Correct. The accumulation of configuration would have issue with frr-reload.

Reply:
@dwalton76 @chiragshah6  Thanks for the clarification!